File: frontend/src/missions.ts
import { HelpCircle, Scale, Link2, PenLine, type LucideIcon } from 'lucide-react'

// DB의 mission_records.mission_type check 제약과 같은 값이어야 한다.
export type MissionType = 'question' | 'rebuttal' | 'connection' | 'expression'

export type Mission = {
  type: MissionType
  label: string
  prompt: string
  color: string
  icon: LucideIcon
}

export const MISSIONS: Mission[] = [
  {
    type: 'question',
    label: '질문',
    prompt: '이 글의 핵심 주장은 뭐지?',
    color: 'var(--hl-question)',
    icon: HelpCircle,
  },
  {
    type: 'rebuttal',
    label: '반박',
    prompt: '이 주장에 반대한다면?',
    color: 'var(--hl-rebut)',
    icon: Scale,
  },
  {
    type: 'connection',
    label: '연결',
    prompt: '내 상황이나 프로젝트와 연결해보면?',
    color: 'var(--hl-connect)',
    icon: Link2,
  },
  {
    type: 'expression',
    label: '표현',
    prompt: '이 글이 놓친 관점은 뭐지?',
    color: 'var(--hl-express)',
    icon: PenLine,
  },
]

export function pickRandomMission(): Mission {
  return MISSIONS[Math.floor(Math.random() * MISSIONS.length)]
}

export function pickRandomSentence(sentences: string[]): string {
  return sentences[Math.floor(Math.random() * sentences.length)]
}

File: backend/app/missions.py
"""서버가 관리하는 미션 타입과 고정 문구.

`GET /api/articles/{articleId}`와 `POST /api/mission-records`가 같은 정의를 공유한다.
"""

from typing import Literal

MissionType = Literal["question", "rebuttal", "connection", "expression"]

MISSION_PROMPTS: dict[MissionType, str] = {
    "question": "이 글의 핵심 주장은 뭐지?",
    "rebuttal": "이 주장에 반대한다면?",
    "connection": "내 상황이나 프로젝트와 연결해보면?",
    "expression": "이 글이 놓친 관점은 뭐지?",
}

RECOMMENDED_MISSION_TYPE: MissionType = "connection"
